app/src/main/java/com/hedvig/app/feature/keygear/ui/itemdetail/KeyGearItemDetailActivity.kt

```diff
@@ -2,13 +2,107 @@ package com.hedvig.app.feature.keygear.ui.itemdetail
 
 import android.content.Context
 import android.content.Intent
+import android.graphics.Color
+import android.os.Bundle
+import androidx.core.view.doOnNextLayout
+import androidx.core.view.updatePadding
+import androidx.core.widget.NestedScrollView
+import com.hedvig.android.owldroid.fragment.KeyGearItemFragment
+import com.hedvig.android.owldroid.graphql.KeyGearItemQuery
+import com.hedvig.android.owldroid.type.KeyGearItemCategory
 import com.hedvig.app.BaseActivity
 import com.hedvig.app.R
+import com.hedvig.app.util.boundedColorLerp
+import com.hedvig.app.util.extensions.compatColor
+import com.hedvig.app.util.extensions.compatDrawable
+import com.hedvig.app.util.extensions.observe
+import com.hedvig.app.util.extensions.view.useEdgeToEdge
+import dev.chrisbanes.insetter.doOnApplyWindowInsets
+import kotlinx.android.synthetic.main.activity_key_gear_item_detail.*
+import org.koin.android.viewmodel.ext.android.viewModel
 
-// TODO: Exit animation. Should be a shared element transition to the newly created item on the tab screen
 class KeyGearItemDetailActivity : BaseActivity(R.layout.activity_key_gear_item_detail) {
 
+    private val model: KeyGearItemDetailViewModel by viewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        supportPostponeEnterTransition()
+
+        root.useEdgeToEdge()
+        initializeToolbar()
+        initializePhotos(
+            intent.getStringExtra(FIRST_PHOTO_URL),
+            intent.getSerializableExtra(CATEGORY) as? KeyGearItemCategory
+        )
+
+        model.data.observe(this) { data ->
+            data?.let { bind(it) }
```

Comment:
shouldn't we log if it is null?

Reply:
There will always be one `null` emitted when the ViewModel is created, so I think we should not 🙂